src/Akeneo/Pim/Enrichment/Component/Product/Command/GroupProductsHandler.php

```diff
@@ -18,58 +19,51 @@
  */
 class GroupProductsHandler
 {
-    private FindProductIdentifiersInterface $findGroupProductIdentifiers;
-    private GroupRepositoryInterface $groupRepository;
-    private BulkSaverInterface $productSaver;
-    private ProductRepositoryInterface $productRepository;
-    private int $batchSize;
-
-
-    public function __construct(FindProductIdentifiersInterface $getGroupProductIdentifiers, GroupRepositoryInterface $groupRepository, BulkSaverInterface $productSaver, ProductRepositoryInterface $productRepository, int $batchSize)
-    {
-        $this->findGroupProductIdentifiers = $getGroupProductIdentifiers;
-        $this->groupRepository = $groupRepository;
-        $this->productSaver = $productSaver;
-        $this->productRepository = $productRepository;
-        $this->batchSize = $batchSize;
+    public function __construct(
+        private FindProductUuidsInGroup $findUuids,
+        private GroupRepositoryInterface $groupRepository,
+        private BulkSaverInterface $productSaver,
+        private ProductRepositoryInterface $productRepository,
+        private int $batchSize
+    ) {
     }
 
     public function handle(GroupProductsCommand $updateProductsToGroupCommand)
     {
-        $newProductIds = $updateProductsToGroupCommand->productIds();
-        $oldProductIds = $this->findGroupProductIdentifiers->fromGroupId($updateProductsToGroupCommand->groupId());
+        $newProductIds = $updateProductsToGroupCommand->productUuids();
```

Comment:
```suggestion
        $newProductUuids = $updateProductsToGroupCommand->productUuids();
``` ? and same on next line